Updated Getting Started Guide for vSphere Cloud Provider

divyenpatel opened the pull request:
This PR includes changes for the Getting Started Guide for vSphere Cloud Provider.

Following is the summary of change.

1. Added refined roles and permissions for vSphere Cloud Provider user.
2. Clarified steps for enabling vSphere Cloud Provider.
3. Removed deprecated content.

Please review : @BaluDontu @luomiao @tusharnt

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4418)
<!-- Reviewable:end -->

divyenpatel commented:
rebased to sync with master branch. 
@jbeda @erictune Can you help with merging this PR

luomiao commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -40,201 +40,114 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
+**[ Step 4 ]** Create and assign Roles to the vSphere Cloud Provider user and vSphere entities.
 
-The recommended version for kube-up is [v1.4.7](https://github.com/kubernetes/kubernetes/releases/tag/v1.4.7)
+vSphere Cloud Provider requires the following minimal set of privileges to interact with vCenter. Please refer [vSphere Documentation Center](http://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.vsphere.security.doc%2FGUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html&resultof=%22%43%72%65%61%74%65%22%20%22%63%72%65%61%74%22%20%22%43%75%73%74%6f%6d%22%20%22%63%75%73%74%6f%6d%22%20%22%52%6f ...
```
This part is kind of complicated...
Is there a way to create a super user directly to include all the privileges, or it has to be done like the the following, to assign all those roles/privileges? Just curious...

divyenpatel replied:
Provided a Note:
`Note: if you want to use Administrator account then this step can be skipped.`

luomiao commented on docs/getting-started-guides/vsphere.md:
````diff
@@ -40,201 +40,114 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
-
-```shell
-govc datastore.ls ./kube/
+Sample Config:
````
For worker nodes, this config could be different.
If worker nodes set vm-name in the config file, they can exclude user/password in their config files.

divyenpatel replied:
Provided two different conf file. and added note

`Note: vm-name parameter is introduced in 1.6.4 release. Both vm-uuid and vm-name are optional parameters. if vm-name is specified then vm-uuid is not used. if both are not specified then kubelet will get vm-uuid from /sys/class/dmi/id/product_serial and query vCenter to find the Node VM's name.`

divyenpatel commented:
Addressed @luomiao review comments.

divyenpatel commented:
I can not debug why `deploy/netlify` has failed.  I do not have access to `/sites/kubernetes-io-master-staging/deploys/59729197a700c44f9d5d1230 `

@jbeda @erictune can you help?

bradbeam commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -40,201 +40,137 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
+        user = "vCenter username for cloud provider"
+        password = "password"
+        server = "IP/FQDN for vCenter"
+        port = "443" #Default 443 for vCenter
```
Isn't the `port` parameter being deprecated?

divyenpatel replied:
Yes it is deprecated and not required in the vsphere.conf file. Removed it.

bradbeam commented on docs/getting-started-guides/vsphere.md:
```diff
@@ -10,7 +10,7 @@ This page covers how to get started with deploying Kubernetes on vSphere and det
 {:toc}
 
-### Getting started with vSphere
+### Getting started with the vSphere
```
This seems incomplete, maybe `Getting started with the vSphere Cloud Provider`?

divyenpatel replied:
fixed.

luomiao commented:
LGTM

jbeda commented on docs/getting-started-guides/vsphere.md:
````diff
@@ -40,201 +40,135 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
-```
-
-* Get or build a [binary release](/docs/getting-started-guides/binary_release)
+vSphere Cloud Provider requires the following minimal set of privileges to interact with vCenter. Please refer [vSphere Documentation Center](http://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.vsphere.security.doc%2FGUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html&resultof=%22%43%72%65%61%74%65%22%20%22%63%72%65%61%74%22%20%22%43%75%73%74%6f%6d%22%20%22%63%75%73%74%6f%6d%22%20%22%52%6f ...
````
Nit: this URL is ugly.  It seems to redirect to a much prettier: https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.vsphere.security.doc/GUID-18071E9A-EED1-4968-8D51-E0B4F526FDA3.html

divyenpatel replied:
Fixed.

jbeda commented on docs/getting-started-guides/vsphere.md:
````diff
@@ -40,201 +40,135 @@ guide](/docs/concepts/storage/volumes/#vspherevolume)
+**`vsphere.conf` for Master Node:**
+ ```
+[Global]
+        user = "vCenter username for cloud provider"
````
I assume that these are values that users can fill in.  Perhaps some sort of indication there like curly braces or something?

jbeda commented:
/lgtm

Not sure what is going on with netlify.  Some nits on style but nothing critical.

chenopis commented:
@divyenpatel It's definitely the `<br>` tag in the table that is causing the error during the preview build. I'm still researching a workaround.

divyenpatel commented:
@chenopis Can we use `\s\s` in the Markdown instead of `<br>` tag as mentioned here - http://markdown-guide.readthedocs.io/en/latest/basics.html?highlight=break#line-return

chenopis commented:
@divyenpatel It didn't seem to work for me when I tried it. I ended up just converting the table into HTML and using the <br> tag. You can preview the changes here: https://deploy-preview-4418--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/vsphere/

chenopis commented:
@divyenpatel I fixed some other formatting issues, so if the preview looks good to you, I can go ahead and merge this.

divyenpatel commented:
Thanks @chenopis for fixing md formatting issues. Preview looks good to me. I have squashed all commits and added known issues section. You can merge this PR.
